### PR TITLE
MoveMovableSubsystem command

### PR DESCRIPTION
**Pull request by OhadLevine**

### CODE REVIEW

**Comment by OhadLevine** [comment]
[Command: MoveMoveableSubsystem](https://trello.com/c/bYtQmNtk/114-command-movemoveablesubsystem)

File: src/main/java/frc/robot/commands/MoveMovableSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.function.BooleanSupplier;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.MoveableSubsystem;
+
+public class MoveMovableSubsystem extends CommandBase {
+  private MoveableSubsystem subsystem;
+  private BooleanSupplier isFinished;
+  private double power;
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/MoveMovableSubsystem.java:12]: private double power;
make supplier

File: src/main/java/frc/robot/commands/MoveMovableSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  public MoveMovableSubsystem(MoveableSubsystem subsystem, double power, BooleanSupplier isFinished) {
+    this.subsystem = subsystem;
+    this.power = power;
+    this.isFinished = isFinished;
+    addRequirements(subsystem);
+  }
+  
+  /** This command can move any subsystem that implements MovableSubsytem without an isFinished */
+  public MoveMovableSubsystem(MoveableSubsystem subsystem, double power) {
+    this(subsystem, power, () -> false);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/MoveMovableSubsystem.java]: this constructor first then the second

File: src/main/java/frc/robot/commands/MoveMovableSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.function.BooleanSupplier;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.MoveableSubsystem;
+
+public class MoveMovableSubsystem extends CommandBase {
+  private MoveableSubsystem subsystem;
+  private BooleanSupplier isFinished;
+  private double power;
+
+  /** This command can move any subsystem that implements MovableSubsytem. */
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/MoveMovableSubsystem.java]: /** This command can move any subsystem that implements MovableSubsytem. */
this constructor uses BooleanSupplier for... etc.

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
few things

File: src/main/java/frc/robot/commands/MoveMovableSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package frc.robot.commands;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/commands/MoveMovableSubsystem.java]: remove empty line

**Review by EliyaFishman — CHANGES_REQUESTED** [review]

**Comment by OhadLevine** [comment]
fixed a second time

**Review by EliyaFishman — APPROVED** [review]

File: src/main/java/frc/robot/commands/MoveMovableSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import frc.robot.subsystems.MoveableSubsystem;
+
+public class MoveMovableSubsystem extends CommandBase {
+  private MoveableSubsystem subsystem;
+  private BooleanSupplier isFinished;
+  private DoubleSupplier power;
+
+  /**
+   * This command can move any subsystem that implements MovableSubsystem without
+   * an isFinished
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/MoveMovableSubsystem.java]: without an isFinished
really?? I am a programmer who didn't write the code. What should I understand from this??

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
last chance

**Comment by OhadLevine** [comment]
done

**Review by HilelAmiel — APPROVED** [review]

**Pull request merged**